### DISCUSSION TITLE
pagination within comboBox

### DISCUSSION THREAD

**Discussion by jeevan-sg** [twilio-labs/paste#2672] [General] [answered]
Hi, I have a requirement to load more items in a combobox when user reaches to the end of the list. 
Is this something currently supported (https://paste.twilio.design/components/combobox)? 
If yes, can you point me to an example? 
If not, can you advice if this is something in the roadmap?

Or any other pattern you advise on implementing - fetch more items into an comboBox. Thank you

**Comment by nkrantz** [comment, marked as answer]
Hi @jeevan-sg! We don't currently have a lazy loading feature in our Combobox component, however we do plan on adding it soon. The planned work is documented in the ["Future additions" portion of the Multiselect Combobox spec](https://docs.google.com/document/d/14BaGDkfoYddL8vFOQ-fH06AXicR7wu3PZNblW1U8ITs/edit?pli=1#bookmark=id.fkuy1q8yqu07). We'll update this thread when the work is completed. Thanks!

**Reply by nkrantz** [reply]
Jira ticket: https://issues.corp.twilio.com/browse/DSYS-3530

**Reply by jeevan-sg** [reply]
Thank you @nkrantz for the response. 

Just wondering what would be expected behavior (pattern) to load more items into combBox? (I couldn't find one from the doc or ticket shared)
- is it when user reaches the end of list, more items will be fetched?
- or fetch all items but limit number of items to be displayed and provide an option to paginate through the list?

Can you advise? Thanks

**Reply by nkrantz** [reply]
We haven't yet done any digging into the ideal pattern for fetching more combobox items, so we aren't able to give guidance on that until the ticketed work is done. Currently, you'd only be able to pre-fetch all the data and lazily render it, using the virtualized combobox. [You can see an example of a virtualized implementation here](https://paste-storybook.twilio.design/?path=/story/components-combobox--virtualized-combobox).

**Reply by sbarilov** [reply]
@nkrantz the link for virtualized combobox does not work or the story book is broken

**Reply by jeevan-sg** [reply]
@nkrantz, can you point us to a sample implementation in code, if any. Thanks

**Reply by nkrantz** [reply]
Thanks for letting me know! We recently added a Multiselect Combobox and changed the folder structure a bit, which is why that link doesn't work anymore. [Here's the new link to that story](https://paste-storybook.twilio.design/?path=/story/components-combobox-combobox--virtualized-combobox). And [here's the code for that story](https://github.com/twilio-labs/paste/blob/main/packages/paste-core/components/combobox/stories/Combobox.stories.tsx#L147).

**Reply by sahibbajaj** [reply]
Is there any update on when the Combobox will support this feature?
